# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/gnunet/hotfix.patch.cross
# Copyright (C) 2023 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- a/meson.build
+++ b/meson.build
@@ -492,14 +492,16 @@ curl_ssl_check = '''#include <curl/curl.
     }
   '''
 
-result = cc.run(
-    curl_ssl_check,
-    name: 'cURL gnutls check',
-    dependencies: curl_dep,
-)
 private_config.set('curl_gnutls', 0)
-if result.returncode() == 0
-    private_config.set('curl_gnutls', 1)
+if meson.can_run_host_binaries()
+    result = cc.run(
+        curl_ssl_check,
+        name: 'cURL gnutls check',
+        dependencies: curl_dep,
+    )
+    if result.returncode() == 0
+        private_config.set('curl_gnutls', 1)
+    endif
 endif
 
 if cc.has_function('getloadavg') == false
